src/App.jsx

```diff
@@ -14,45 +16,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [activeSort, setActiveSort] = useState(null);
+  const [isReversed, setIsReversed] = useState(false);
+  const reset = 'reset';
+  const alphabeticaly = 'alphabeticaly';
+  const length = 'byLength';
+  const reverse = 'reverse';
+  const isSorted = activeSort !== null;
+
+  const handleSort = (sortType) => {
+    const sortFunctions = {
+      alphabeticaly: () => {
+        const sortedGoods
+          = [...goods].sort((a, b) => a[0].localeCompare(b[0],
+            'en', { sensitivity: 'base' }));
+
+        setGoods(sortedGoods);
+        setActiveSort(sortType);
+        setIsReversed(false);
+      },
+      byLength: () => {
+        const sortedGoods = [...goods].sort((a, b) => a.length - b.length);
+
+        setGoods(sortedGoods);
+        setActiveSort(sortType);
+        setIsReversed(false);
+      },
+      reverse: () => {
+        const reversedGoods = [...goods].reverse();
+
+        setGoods(reversedGoods);
+        setActiveSort(sortType);
+        setIsReversed(!isReversed);
+      },
```

Comment:
reverse should only set opposite isReveses state.
compare working your implementation with working example
it is reason of failed tests.
working example:
![working](https://github.com/mate-academy/react_list-of-goods-js/assets/33029864/1d6c30c8-5b10-42da-bd02-9f273519c175)
actual:
![actual-goods](https://github.com/mate-academy/react_list-of-goods-js/assets/33029864/4eaf5182-b557-4c13-a587-6a7de51f42e2)